ui/shared/components/panel/variants/VariantIndividuals.jsx

```diff
@@ -377,10 +363,166 @@ const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById })
+    if (warning) {
+      acc[genotype.sampleType] = warning
+    }
+    return acc
+  }, {})
+
+  const hasDifferentNumAlt = genotypes.some(genotype => genotype.numAlt !== genotypes[0].numAlt)
+
+  return (
+    <Alleles
+      genotype={genotypes[0]}
+      variant={variant}
+      isHemiX={isHemiX}
+      warning={(Object.keys(warnings).length > 0 || hasDifferentNumAlt) && (
+        <div>
+          {Object.entries(warnings).map(([sampleType, warning]) => (
+            <div key={sampleType}>
+              <b>
+                {sampleType}
+                Warning:
+              </b>
+              {warning}
+            </div>
+          ))}
+          {hasDifferentNumAlt && (
+            <span>
+              <b>Warning: </b>
+              Genotypes differ across sample types
+              {genotypes.map(genotype => (
+                <div key={genotype.sampleType}>
+                  {genotype.sampleType}
+                  :
+                  <AlleleContainer>
+                    <AllelesHeader genotype={genotype} variant={variant} isHemiX={isHemiX} />
+                  </AlleleContainer>
+                </div>
+              ))}
+            </span>
+          )}
+        </div>
+      )}
+    />
+  )
+})
+
+MultiSampleTypeAlleles.propTypes = {
+  genotypes: PropTypes.arrayOf(PropTypes.object),
+  variant: PropTypes.object,
+  individual: PropTypes.object,
+  isHemiX: PropTypes.bool,
+  isCompoundHet: PropTypes.bool,
+}
+
+const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById }) => {
+  const individualGenotypes = variant.genotypes[individual.individualGuid]
+  if (!individualGenotypes) {
+    return null
+  }
+  const genotypes = (Array.isArray(individualGenotypes) ? individualGenotypes : [individualGenotypes]).sort(
+    (a, b) => SAMPLE_TYPE_DISPLAY_ORDER.indexOf(a.sampleType) - SAMPLE_TYPE_DISPLAY_ORDER.indexOf(b.sampleType),
+  )
+
+  const isHemiX = isHemiXVariant(variant, individual)
+
+  const allDetails = genotypes.map(genotype => genotypeDetails(genotype, variant, genesById)).reduce(
+    (acc, genotypeDetail, index) => {
+      if (index > 0) {
+        acc.push(<Divider />)
+      }
+      return acc.concat(genotypeDetail)
+    }, [],
+  )
+
+  const content = (
+    <span>
+      {genotypes.length === 1 ? (
+        <span>
+          <LegacyAdditionalSampleTypePopup
+            genotype={genotypes[0]}
+            variant={variant}
+            isHemiX={isHemiX}
+            genesById={genesById}
+          />
+          <Alleles
+            genotype={genotypes[0]}
+            variant={variant}
+            isHemiX={isHemiX}
+            warning={
+              <div>
+                <b>Warning: </b>
+                {getWarningsForGenotype(genotypes[0], variant, individual, isHemiX, isCompoundHet)}
+              </div>
+            }
+          />
+        </span>
+      ) : (
+        <MultiSampleTypeAlleles
```

Comment:
So in the end `MultiSampleTypeAlleles` returns `Alleles` using the same props as for a single-sample genotype - the only difference is the warning. Rather than have a helper class and a ternary here, it would be much neater to have a `getWarningsForGenotypes` method that handles constructing the correct warning for all the genotypes and leave this rendering much simpler

Reply:
I removed `MultiSampleTypeAlleles` and conditionally set the `warning` prop on `Alleles` based on genotypes length.